core/src/main/java/org/frankframework/http/rest/ApiListener.java

```diff
@@ -169,6 +169,11 @@ public void close() {
 
 	@Override
 	public Message processRequest(Message message, PipeLineSession session) throws ListenerException {
+		if (getMethod() == HttpMethod.HEAD) {
+			// If it's a HEAD request, return an empty message
+			return new Message("");
```

Comment:
I guess it should be returning no data, instead of an empty message

Reply:
I think it is wrong that we return an empty message instead of NULL, because below we say that we explicitly want NULL returned instead of empty messages.

In addition, I think it is wrong that we skip execution here altogether. I mean, if we do not execute code in this method, which is called from ApiListenerServlet, why even invoke it altogether?

I would rather have the listener executed normally so the adapter / pipeline has a change to process the HEAD request in a way it sees fit. For instance, to correctly determine the actual content-type header for the response (which could be different from the default configured on the ApiListener).

@jjansenvr @alisihab What do you think of this? Should HTTP method HEAD invoke the pipeline like a normal call in order to potentially compute content-type (and perhaps content-length?) header and let the pipeline itself figure out what to do with method HEAD?

Or should pipeline execution be skipped, and default content-type be returned?

(I'm thinking of the case of a pipeline that downloads something like an image. The content-type will be the particular image-type: JPEG, PNG, etc. So content-type depends on the result).

Reply:
I discussed my questions on this with Jeroen and we concluded that the full pipeline should be executed as normal. So we should not skip execution and just return an empty message.

The ApiListenerServlet should then apply all headers including content-length and content-type as normal for any request, but should not return the actual body.

Jeroen has updated the ticket with the functional requirements regarding this.